sfs/time/drivingfunction.py

```diff
@@ -1,64 +1,197 @@
 
     """
     if c is None:
         c = defs.c
-    if fs is None:
-        fs = defs.fs
     x0 = util.asarray_of_rows(x0)
-    nx0 = util.asarray_of_rows(nx0)
+    n0 = util.asarray_of_rows(n0)
+    n = util.asarray_1d(n)
+    xref = util.asarray_1d(xref)
+    g0 = np.sqrt(2 * np.pi * np.linalg.norm(xref - x0, axis=1))
+    delays = inner1d(n, x0)/c
+    weights = 2 * g0 * inner1d(n, n0)
+    return delays, weights
+
+
+def wfs_25d_point(x0, n0, xs, xref=[0, 0, 0], c=None):
+    r"""Point source by 2.5-dimensional WFS.
+
+    Parameters
+    ----------
+    x0 : (N, 3) array_like
+        Sequence of secondary source positions.
+    n0 : (N, 3) array_like
+        Sequence of secondary source orientations.
+    xs : (3,) array_like
+        Virtual source position.
+    xref : (3,) array_like, optional
+        Reference position
+    c : float, optional
+        Speed of sound
+
+    Returns
+    -------
+    delays : (N,) numpy.ndarray
+        Delays of secondary sources in seconds.
+    weights: (N,) numpy.ndarray
+        Weights of secondary sources.
+
+    Notes
+    -----
+    2.5D correction factor
+
+    .. math::
+
+         g_0 = \sqrt{2 \pi |x_\mathrm{ref} - x_0|}
+
+
+    d using a point source as source model
+
+    .. math::
+
+         d_{2.5D}(x_0,t) = h(t)
+         \frac{g_0  \scalarprod{(x_0 - x_s)}{n_0}}
+         {2\pi |x_0 - x_s|^{3/2}}
+         \dirac{t - \frac{|x_0 - x_s|}{c}}
+
+    with wfs(2.5D) prefilter h(t), which is not implemented yet.
+
+    References
+    ----------
+    See http://sfstoolbox.org/en/latest/#equation-d.wfs.ps.2.5D
+
+    """
+    if c is None:
+        c = defs.c
+    x0 = util.asarray_of_rows(x0)
+    n0 = util.asarray_of_rows(n0)
     xs = util.asarray_1d(xs)
-    g0 = np.sqrt(2*np.pi*np.linalg.norm(xref-x0, axis=1))
+    xref = util.asarray_1d(xref)
+    g0 = np.sqrt(2 * np.pi * np.linalg.norm(xref - x0, axis=1))
     ds = x0 - xs
-    r = np.linalg.norm(x0-xs, axis=1)
-    delay = 1/c * r
-    weight = g0/(2*np.pi) * inner1d(ds, nx0) / r**(3/2)
-    sig = np.ones(len(x0))  # dirac
-    line = delayline(sig, delay*fs, weight)
-    return delay, weight, line
+    r = np.linalg.norm(x0 - xs, axis=1)
+    delays = r/c
+    weights = g0/(2 * np.pi) * inner1d(ds, n0) / r**(3/2)
+    return delays, weights
+
+
+def driving_signals(delays, weights, signal, fs=None):
+    """Get driving signals per secondary source.
+
+    Returned signal is the delayed and weighted input signal per channel.
+
+    Parameters
+    ----------
+    delays : (C,) array_like
+        Delay in seconds for each channel, negative values allowed.
+    weights : (C,) array_like
+        Amplitude weighting factor for each channel.
+    signal : (N,) array_like
+        Excitation signal (mono) which gets weighted and delayed.
+    fs: int, optional
+        Sampling frequency in hertz.
+
+    Returns
+    -------
+    (N, C) numpy.ndarray
+        N is signal samples(t) per channel C, delayed and weighted.
```

Comment:
I don't understand this sentence. What does "t" mean?

I think you should move the explanation of C and N up, because those are used before.